### DISCUSSION TITLE
Error with MSD and SDC

### DISCUSSION THREAD

**Discussion by jmarbues** [brucefan1983/GPUMD#726] [Q&A] [answered]
Hi. I was trying to run some simulations to calculate the self-diffusion coefficient via the velocity autocorrelation and the mean-square displacement function. In both approaches the simulation immediately crashes with the following error:

CUDA Error:                       
File:       utilities/gpu_vector.cuh                                                                                    
Line:       83                                                                                                          
Error code: 2                                                                                                         
Error text: out of memory

Thank you for your time. 
Kind regards, José

![msd_error](https://github.com/user-attachments/assets/82574af3-5ac8-40c4-b9c0-88fec36b1345)
![sdc_error](https://github.com/user-attachments/assets/b55c86f1-5b41-40b9-8ca2-6f853d1508a3)

**Comment by brucefan1983** [comment, marked as answer]
You number of correlation stpes is 5 000 000, which is too large. Your sampling inverval is 5. If your time step is 1 fs, this means your maximum correlation time is 25 ns. I have never seen a diffusion problem requiring so large correlation time. You can consider decreasing the number of correlation steps, and if needed, increasing the sampling interval. 

Note: 
The memory required for the calculations is at least `N*5000000*24` Byte based on your inputs, where `N` is the number of atoms

**Comment by jmarbues** [comment]
Ok yes, that solved the problem. Thank you for your help. 

The formula for the memory required you gave is very useful.
